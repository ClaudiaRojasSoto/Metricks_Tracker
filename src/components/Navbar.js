import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <nav>
    <NavLink to="/" exact="true" activeclassname="active">Home</NavLink>
    {' '}
    <NavLink to="/details/1" activeclassname="active">Details</NavLink>
  </nav>
);

export default Navbar;
