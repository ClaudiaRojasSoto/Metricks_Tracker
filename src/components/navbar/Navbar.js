import React from 'react';
import { NavLink } from 'react-router-dom';
import { FiMic, FiSettings } from 'react-icons/fi';
import s from './style.module.css';

const Navbar = () => (
  <nav className={s.navbar}>
    <NavLink to="/" exact="true" activeclassname="active">Home</NavLink>
    {' '}
    <NavLink to="/details/1" activeclassname="active">Details</NavLink>
    <FiMic />
    <FiSettings />
  </nav>
);

export default Navbar;
