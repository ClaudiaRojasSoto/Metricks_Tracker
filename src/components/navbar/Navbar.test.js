import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Navbar from './Navbar';

describe('Navbar tests', () => {
  let navbar;
  beforeAll(() => {
    navbar = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
  });

  it('Component should render correctly', () => {
    expect(navbar).toBeTruthy();
  });

  it('It should match its snapshot', () => {
    expect(navbar).toMatchSnapshot();
  });

  describe('Back button', () => {
    it('should render back button if not on the root path', () => {
      window.history.pushState({}, '', '/not-root');
      render(
        <Router>
          <Navbar />
        </Router>,
      );
      const backButton = screen.getByRole('button');
      expect(backButton).toBeInTheDocument();
    });

    it('should not render back button on the root path', () => {
      window.history.pushState({}, '', '/');
      render(
        <Router>
          <Navbar />
        </Router>,
      );
      const backButton = screen.queryByRole('button');
      expect(backButton).not.toBeInTheDocument();
    });
  });
});
