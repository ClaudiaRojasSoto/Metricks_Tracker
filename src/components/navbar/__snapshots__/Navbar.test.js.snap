// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests It should match its snapshot 1`] = `
<nav
  className="navbar"
>
  <svg
    className="mic_icon"
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
    />
    <path
      d="M19 10v2a7 7 0 0 1-14 0v-2"
    />
    <line
      x1="12"
      x2="12"
      y1="19"
      y2="23"
    />
    <line
      x1="8"
      x2="16"
      y1="23"
      y2="23"
    />
  </svg>
  <svg
    className="settings_icon"
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="3"
    />
    <path
      d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
    />
  </svg>
</nav>
`;
